src/service_provider/service_provider.routes.js

```diff
@@ -0,0 +1,224 @@
+ *       200:
+ *         description: Successful
+ *         schema:
+ *            $ref: '#/definitions/ServiceProvider'
+ */
+router.get('/:id', async (req, res) => {
+    try {
+        let data = await FindAllServiceProvidersById(req.db, req.params)
+        res.json(data);
+    } catch (err) {
+        res.status(500).json({error: "There was a server error"})
+    }
+});
+
+/**
+ * @swagger
+ *
+ * /service_provider:
+ *   post:
+ *     description: Service Provider create one route
+ *     produces:
+ *       - application/json
+ *     parameters:
+ *       - in:  body
+ *         description: Service Provider Updated Data Object
+ *         required: true
+ *         type: string
+ *         schema:
+ *           $ref: '#/definitions/NewServiceProvider'
+ *     responses:
+ *       500:
+ *         description: Server Error
+ *       200:
+ *         description: Successful
+ *         schema:
+ *            $ref: '#/definitions/ServiceProvider'
+ */
+router.post('/', async (req, res) => {
+    try {
+        let data = await CreateServiceProvider(req.db, req.body)
+        res.json(data);
+    } catch (err) {
+        res.status(500).json({error: "There was a server error"})
+    }
+});
+
+/**
+ * @swagger
+ *
+ * /service_provider/:id:
+ *   put:
+ *     description: Service Provider Update One Route
+ *     produces:
+ *       - application/json
+ *     parameters:
+ *       - in: path
+ *         name: id
+ *         type: string
+ *         required: true
+ *         description: Id of the service provider requesting
+ *       - in:  body
+ *         description: Service Provider Updated Data Object
+ *         required: true
+ *         type: string
+ *         schema:
+ *           $ref: '#/definitions/ServiceProviderUpdate'
+ *     responses:
+ *       500:
+ *         description: Server Error
+ *       200:
+ *         description: Successful
+ *         schema:
+ *            $ref: '#/definitions/ServiceProvider'
+ */
+router.put('/:id', async (req, res) => {
+    try {
+        let data = await UpdateServiceProvider(req.db, req.params, req.body)
+        res.json(data);
+    } catch (err) {
+        res.status(500).json({error: "There was a server error"})
+    }
+});
+
+/**
+ * @swagger
+ *
+ * /service_provider/:id:
+ *   delete:
+ *     description: Service Provider Delete One Route
+ *     produces:
+ *       - application/json
+ *     parameters:
+ *       - in: path
+ *         name: id
+ *         type: string
+ *         required: true
+ *         description: Id of the service provider requesting to delete
+ *     responses:
+ *       500:
+ *         description: Server Error
+ *       204:
+ *         description: Successful
+ */
+router.delete('/:id', async (req, res) => {
+    try {
+        await DeleteServiceProvider(req.db, req.params)
+        res.status(204).json({success: true});
+    } catch (err) {
+        res.status(500).json({error: "There was a server error"})
+    }
+});
+
+module.exports = router;
```

Comment:
I honestly don't love how much noise the swagger/openapi annotations add to these route files. Just a fact of life I guess.